src/AppInstallerCommonCore/Fonts.cpp

```diff
@@ -172,4 +958,184 @@ namespace AppInstaller::Fonts
+                                    if (scope == Manifest::ScopeEnum::Machine)
+                                    {
+                                        // Capture the filename for winget machine fonts so we can avoid duplicates.
+                                        wingetMachineFonts[filePath.filename()] = true;
+                                    }
+
+                                    auto context = FontContext();
+                                    context.Scope = scope;
+                                    context.InstallerSource = InstallerSource::WinGet;
+                                    context.PackageId = packageId;
+                                    context.PackageVersion = version;
+                                    auto fontFile = CreateFontFileInfo(context, filePath, ConvertToUTF16(versionValue.Name()));
+                                    fontFiles.push_back(std::move(fontFile));
+                                }
+                            }
+                        }
+                    }
+                }
+
+                // All remaining values in the root are externally installed fonts.
+                for (const auto& rootValue : root.Values())
+                {
+                    auto value = rootValue.Value();
+                    if (!value.has_value() || (value.value().GetType() != Registry::Value::Type::String))
+                    {
+                        continue;
+                    }
+
+                    std::filesystem::path filePath = { value->GetValue<Registry::Value::Type::String>() };
+                    if (scope == Manifest::ScopeEnum::Machine)
+                    {
+                        // Skip it if this is already identified in the WinGet machine fonts.
+                        if (wingetMachineFonts[filePath.filename()])
+                        {
+                            continue;
+                        }
+                    }
+
+                    auto context = FontContext();
+                    context.Scope = scope;
+                    context.InstallerSource = InstallerSource::Unknown;
+                    auto fontFile = CreateFontFileInfo(context, filePath, ConvertToUTF16(rootValue.Name()));
+                    fontFiles.push_back(std::move(fontFile));
+                }
+            }
+        }
+        catch (...)
+        {
+            LOG_CAUGHT_EXCEPTION();
+            AICLI_LOG(Core, Error, << L"Failed getting font file information.");
+        }
+
+        return fontFiles;
+    }
+
+    std::vector<FontPackageInfo> GetInstalledFontPackages(Manifest::ScopeEnum scope)
+    {
+        auto fontPackages = std::vector<FontPackageInfo>();
+
+        try
+        {
+            auto hive = scope == Manifest::ScopeEnum::Machine ? HKEY_LOCAL_MACHINE : HKEY_CURRENT_USER;
+            auto root = Registry::Key::OpenIfExists(hive, s_FontsPathSubkey.data(), 0UL, KEY_ALL_ACCESS);
```

Comment:
```suggestion
            auto root = Registry::Key::OpenIfExists(hive, s_FontsPathSubkey.data(), 0UL, KEY_READ);
```
?

Reply:
Fix this and in another place where only READ_KEY is necessary.